app/scripts/app.js

```diff
@@ -21,14 +21,34 @@ taskList.config(["$stateProvider", "$locationProvider", function($stateProvider,
 taskList.controller("MainController.controller", ["$scope", "$firebaseArray", function($scope, $firebaseArray) {
 
   var ref = new Firebase("https://luminous-fire-9311.firebaseio.com/messages");
-  $scope.messages = $firebaseArray(ref);
+  $scope.tasks = $firebaseArray(ref);
+  var intervalID;
+
+  $scope.addTask = function() {
+    var time = new Date();
+    $scope.tasks.$add({
+      desc: $scope.newTaskDescription,
```

Comment:
It's an early state of the app so I didn't check if I have valid information.It need to be done indeed.

But I have no idea how to "block malicious code".